barretenberg/cpp/src/barretenberg/dsl/acir_proofs/acir_composer.hpp

```diff
@@ -32,13 +32,23 @@ class AcirComposer {
     bool verify_proof(std::vector<uint8_t> const& proof);
 
     std::string get_solidity_verifier();
-    size_t get_total_circuit_size() { return builder_.get_total_circuit_size(); };
-    size_t get_dyadic_circuit_size() { return builder_.get_circuit_subgroup_size(builder_.get_total_circuit_size()); };
+    size_t get_finalized_total_circuit_size() { return builder_.get_finalized_total_circuit_size(); };
```

Comment:
two new functions that give a finalized size (and check that the circuit is actually finalized)